dbms/src/Interpreters/PredicateExpressionsOptimizer.cpp

```diff
@@ -398,6 +398,9 @@ ASTs PredicateExpressionsOptimizer::evaluateAsterisk(ASTSelectQuery * select_que
                 storage = context.tryGetTable(database_and_table_name.first, database_and_table_name.second);
             }
 
+            if (!storage)
```

Comment:
Throw an exception is better?  And I have similar commit in #3400

Reply:
Well. I see it.